cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -733,7 +734,10 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
                         if (Annotations.booleanAnnotation(sts, Annotations.ANNO_STRIMZI_IO_MANUAL_ROLLING_UPDATE,
                                 false, Annotations.ANNO_OP_STRIMZI_IO_MANUAL_ROLLING_UPDATE)) {
                             return kafkaSetOperations.maybeRollingUpdate(sts, pod -> {
-
+                                if (pod == null) {
+                                    log.debug("Pod not found, no need to roll");
+                                    throw new ConcurrentDeletionException("Pod not found, no need to roll");
```

Comment:
I think you should adapt the message both here and the log. It does not make sense in relation to the exception which essentially kills the reocnciliation.

Reply:
@stanlyDoge think of this more like a failed assertion: If we got here it means something relatively unexpected happened, because the pod must have got deleted since we started the reconciliation loop. So something like "Unexpectedly pod no longer exists during roll of StatefulSet" would be better.